packages/plugins/response-cache-redis/test/response-redis-cache.spec.ts

```diff
@@ -0,0 +1,1155 @@
+          }
+        }
+      }
+    `;
+
+    // query and cache
+    const queryResult = await testInstance.execute(query);
+
+    let cacheHitMaybe = queryResult['extensions']['responseCache']['hit'];
+    expect(cacheHitMaybe).toBeFalsy();
+
+    // get from cache
+    const cachedResult = await testInstance.execute(query);
+
+    cacheHitMaybe = cachedResult['extensions']['responseCache']['hit'];
+    expect(cacheHitMaybe).toBeTruthy();
+
+    const mutationResult = await testInstance.execute(
+      /* GraphQL */ `
+        mutation test($id: ID!) {
+          updateUser(id: $id) {
+            id
+          }
+        }
+      `,
+      {
+        id: 1,
+      }
+    );
+
+    cacheHitMaybe = mutationResult['extensions']['responseCache']['hit'];
+    expect(cacheHitMaybe).toBeFalsy();
+  });
+
+  test('should purge cache on mutation and include invalidated entities', async () => {
+    const spy = jest.fn(() => [
+      {
+        id: 1,
+        name: 'User 1',
+        comments: [
+          {
+            id: 1,
+            text: 'Comment 1 of User 1',
+          },
+        ],
+      },
+      {
+        id: 2,
+        name: 'User 2',
+        comments: [
+          {
+            id: 2,
+            text: 'Comment 2 of User 2',
+          },
+        ],
+      },
+    ]);
+    const schema = makeExecutableSchema({
+      typeDefs: /* GraphQL */ `
+        type Query {
+          users: [User!]!
+        }
+
+        type Mutation {
+          updateUser(id: ID!): User!
+        }
+
+        type User {
+          id: ID!
+          name: String!
+          comments: [Comment!]!
+          recentComment: Comment
+        }
+
+        type Comment {
+          id: ID!
+          text: String!
+        }
+      `,
+      resolvers: {
+        Query: {
+          users: spy,
+        },
+        Mutation: {
+          updateUser(_, { id }) {
+            return {
+              id,
+            };
+          },
+        },
+      },
+    });
+
+    const testInstance = createTestkit([useResponseCache({ cache, includeExtensionMetadata: true })], schema);
+
+    const query = /* GraphQL */ `
+      query test {
+        users {
+          id
+          name
+          comments {
+            id
+            text
+          }
+        }
+      }
+    `;
+
+    const result = await testInstance.execute(
+      /* GraphQL */ `
+        mutation test($id: ID!) {
+          updateUser(id: $id) {
+            id
+          }
+        }
+      `,
+      {
+        id: 1,
+      }
+    );
+
+    const responseCache = result['extensions']['responseCache'];
+    const invalidatedEntities = responseCache['invalidatedEntities'];
+    expect(invalidatedEntities).toHaveLength(1);
+
+    const invalidatedUser = invalidatedEntities[0];
+    expect(invalidatedUser).toHaveProperty('typename');
+    expect(invalidatedUser).toHaveProperty('id');
+    expect(invalidatedUser.typename).toEqual('User');
+    expect(invalidatedUser.id).toEqual('1');
```

Comment:
instead of all those seperate assertions this would be a shorter one that covers all:

```
expect(invalidatedUser).toEqual({
  typename: "User",
  id: "1"
})
```